engines/script/engine.go

```diff
@@ -0,0 +1,70 @@
+package scriptengine
+
+import (
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+	schematypes "github.com/taskcluster/go-schematypes"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+type engineProvider struct {
+	engines.EngineProviderBase
+}
+
+type engine struct {
+	engines.EngineBase
+	Log         *logrus.Entry
+	config      configType
+	schema      schematypes.Object
+	environment *runtime.Environment
+}
+
+func init() {
+	engines.Register("script", engineProvider{})
+}
+
+func (engineProvider) ConfigSchema() schematypes.Schema {
+	return configSchema
+}
+
+func (engineProvider) NewEngine(options engines.EngineOptions) (engines.Engine, error) {
+	var config configType
+	if schematypes.MustMap(configSchema, options.Config, &config) != nil {
```

Comment:
No, payload and config options are validated against the schema before being passed in here...
The validation part of map is therefor unnecessary, but nice. It enforces the contract that's all..

IMO, we could panic on the error if we wanted to...